src/formats/asyncapi.ts

```diff
@@ -0,0 +1,10 @@
+import { isObject } from 'lodash';
+
+type MaybeAsyncApi2 = Partial<{ asyncapi: unknown }>;
+
+const bearsAStringPropertyNamed = (document: unknown, propertyName: string) => {
+  return isObject(document) && propertyName in document && typeof document[propertyName] === 'string';
+};
+
+export const isAsyncApiv2 = (document: unknown) =>
+  bearsAStringPropertyNamed(document, 'asyncapi') && String((document as MaybeAsyncApi2).asyncapi) === '2.0.0';
```

Comment:
Much like OpenAPI, the patch version should be ignored. 

https://github.com/stoplightio/spectral/blob/develop/src/formats/openapi.ts

```
export const isOpenApiv3 = (document: unknown) =>
  isObject(document) && 'openapi' in document && parseFloat(String((document as MaybeOAS3).openapi)) === 3;
```

Notice how it's just looking for the major "3" and not worried about the minor or patch?

Reply:
@philsturgeon Fixed

Will now accept `2.{minor}.{patch}` format where

`minor` or `patch`:
 - can only be made of numbers
 - can not start with a `0`